### PR TITLE
Pandas 1.3.0 Upgrade

### PR DESCRIPTION
**Pull request by chukarsten**
Addresses issues introduced with Pandas 1.3.0 rc1. #2431 Preliminary work required to prepare for Woodwork 0.5.0.

### CODE REVIEW

File: evalml/data_checks/class_imbalance_data_check.py [hunk 1]
```diff
@@ -91,10 +91,12 @@ def validate(self, X, y):
             error_msg = "The number of instances of these targets is less than 2 * the number of cross folds = {} instances: {}"
             DataCheck._add_message(
                 DataCheckError(
-                    message=error_msg.format(self.cv_folds, below_threshold_values),
+                    message=error_msg.format(
```
Comment (chukarsten) [inline evalml/data_checks/class_imbalance_data_check.py:94]: Something about the new pandas caused the order in the datachecks details to change.  I sorted the ones that got changed so it shouldn't be an issue going forward.

File: evalml/pipelines/components/estimators/estimator.py [hunk 2]
```diff
@@ -62,6 +64,8 @@ def predict(self, X):
         """
         try:
             X = infer_feature_types(X)
```
Comment (chukarsten) [inline evalml/pipelines/components/estimators/estimator.py:66]: For some reason, passing the equivalent RangeIndex to XGBoost's DMatrix results in the columns receiving an appended space when converted to strings.  e.g. ["0 ", "1 ", "2 "...]

**Review by chukarsten — COMMENTED** [review]
The two places that the pandas upgrade from 1.2.5 to 1.3.0 manifested themselves were: 

1. The columns of XGBoost's DMatrix created from the new DataFrame resulted in the integer column labels being prepended extra spaces so that they would be strings with exactly two characters.  Not sure why this happens with the new Pandas, but I think this is a pandas/XGBoost interaction, so might file an issue there.
2. The order, but not the values, of some of the datacheck validation results has changed.  The affected tests are rewritten so the order doesn't matter.

**Review by jeremyliweishih — APPROVED** [review]
nice!

File: evalml/data_checks/class_imbalance_data_check.py (hunk 1 above)
Comment (freddyaboulton) [reply evalml/data_checks/class_imbalance_data_check.py:94]: Must be `value_counts`? It's called in both this data check and InvalidTargetDataCheck

File: evalml/pipelines/components/estimators/estimator.py (hunk 2 above)
Comment (freddyaboulton) [reply evalml/pipelines/components/estimators/estimator.py:66]: Bizzare.

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @chukarsten ! Thanks for doing this.

**Review by bchen1116 — APPROVED** [review]
LGTM!

File: evalml/data_checks/class_imbalance_data_check.py (hunk 1 above)
Comment (angela97lin) [reply evalml/data_checks/class_imbalance_data_check.py:94]: Frustrating, I believe they changed the behavior of `value_counts` not too many releases ago too 🤔

**Review by angela97lin — APPROVED** [review]
Not that you needed another green seal of approval, but LGTM!

File: evalml/pipelines/components/estimators/estimator.py (hunk 2 above)
Comment (chukarsten) [reply evalml/pipelines/components/estimators/estimator.py:71]: Ultimately this was an issue with xgboost and DMatrix, resolved in xgboost > 1.4.2.

**Pull request merged**